aclk/agent_cloud_link.c

```diff
@@ -196,31 +251,50 @@ struct aclk_query *aclk_query_find(char *topic, char *data, char *msg_id, char *
  * Add a query to execute, the result will be send to the specified topic
  */
 
-int aclk_queue_query(char *topic, char *data, char *msg_id, char *query, int run_after, int internal)
+int aclk_queue_query(char *topic, char *data, char *msg_id, char *query, int run_after, int internal, ACLK_CMD aclk_cmd)
 {
     struct aclk_query *new_query, *tmp_query;
 
     // Ignore all commands while we wait for the agent to initialize
     if (unlikely(waiting_init))
         return 0;
 
+    // Ignore all commands if agent not stable and reset the last_init_sequence mark
+    if (agent_state == 0) {
+        last_init_sequence = now_realtime_sec();
+        return 0;
+    }
+
     run_after = now_realtime_sec() + run_after;
 
     QUERY_LOCK;
-    tmp_query = aclk_query_find(topic, data, msg_id, query);
+    struct aclk_query *last_query = NULL;
+
+    //last_query = NULL;
+    tmp_query = aclk_query_find(topic, data, msg_id, query, aclk_cmd, &last_query);
```

Comment:
Removing duplicate queries was not in the spec - what if the far end is trying to reduce latency by pipelining, i.e. issuing multiple queries in sequence so that the req/responses overlap?

Reply:
Will remove / discuss in future issue